java/src/main/java/com/google/oak/client/OakClient.java

```diff
@@ -42,15 +51,24 @@ public class OakClient<T extends Transport> implements AutoCloseable {
    * @return an instance of the {@code OakClient} wrapped in a {@code Result}
    */
   public static <E extends EvidenceProvider & Transport, V extends AttestationVerifier>
-      Result<OakClient<E>, Exception> create(E transport, V verifier) {
+      Result<OakClient<E>, Exception> create(
+          E transport, V verifier, Clock clock, ReferenceValues referenceValues) {
     // TODO(#3641): Implement client-side attestation verification.
     return transport.getEvidence()
         .mapError(Exception::new)
         .andThen(e
-            -> verifier.verify(e.getAttestationEvidence(), e.getAttestationEndorsement())
+            -> verifier
+                   .verify(clock.instant().toEpochMilli(), e.getEvidence(), e.getEndorsements(),
+                       referenceValues)
                    .map(b
                        -> new OakClient<E>(transport,
-                           e.getAttestationEvidence().getEncryptionPublicKey().toByteArray())));
+                           e.hasEvidence() ? e.getEvidence()
+                                                 .getApplicationKeys()
+                                                 .getEncryptionPublicKeyCertificate()
```

Comment:
We also need to unpack the key from the corresponding certificate. I think right now it's already being done by the `verify` function. So the needed key is in the `b.encryptionPublicKey`: https://github.com/project-oak/oak/blob/e54ea4fcbc14ed22c3cffc72ea152910e95b149d/proto/attestation/verification.proto#L43

Reply:
Done